sys/arm64/arm64/trap.c

```diff
@@ -299,11 +299,11 @@ print_registers(struct trapframe *frame)
 
 	for (reg = 0; reg < nitems(frame->tf_x); reg++) {
 		printf(" %sx%d: %16lx\n", (reg < 10) ? " " : "", reg,
-		    frame->tf_x[reg]);
```

Comment:
It would probably be nice to print the cap register for these instead of just the address using an #ifdef as we do on RISC-V.  For now we can disable -Wformat until the right change is cherry-picked, but you can just use CHERI_CAP_PRINT* or even use a macro here for the printf string that uses "%16lx" for aarch64 and "%.16#p" for Morello.

Reply:
If you mean `_CHERI_PRINTF_CAP*` we can't as you broke it on Morello in 06f889ad1f8120fa682db902cf23536a20e593ff. The toolchain needs an update before we can switch.

Reply:
No, you can just turn off -Wformat for now until the toolchain is updated (there's even a specific make knob for that for the kernel due to building with stock GCC).  The code in sys/kern/subr_prf.c that does the actual printf should work fine on Morello.